node/src/reactor/joiner.rs

```diff
@@ -509,8 +509,10 @@ impl reactor::Reactor for Reactor {
 
         let linear_chain = linear_chain::LinearChain::new(
             &registry,
-            chainspec_loader.initial_state_root_hash(),
             protocol_version.clone(),
+            chainspec_loader.initial_state_root_hash(),
+            chainspec_loader.chainspec().core_config.auction_delay,
+            chainspec_loader.chainspec().core_config.unbonding_delay,
```

Comment:
Might be safer to pass in a chainspec reference: Then it's impossible to accidentally switch auction and unbonding delay.

Reply:
I did that initially but then I thought that passing in the whole thing might be problematic when we want to test it/mock it.